numpy/lib/format.py

```diff
@@ -431,6 +433,14 @@ def _write_array_header(fp, d, version=None):
         header.append("'%s': %s, " % (key, repr(value)))
     header.append("}")
     header = "".join(header)
+    
+    # Add spare space for growth but make sure the array header size stays
+    # constant at all times.
+    shape = d['shape']
+    header += " " * ((GROWTH_AXIS_MAX_DIGITS - len(repr(
+        shape[-1 if d['fortran_order'] else 0]
+    ))) if len(shape) > 0 else 0)
```

Comment:
This could probably be simplified with some intermediates or so, but is fine and easy enough to read.

Reply:
I think so, too.